### PR TITLE
Implement the class language element part I.

### PR DESCRIPTION
**Pull request by rerobika**
Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

**Comment by yichoi** [comment]
please enable more TCs at https://github.com/pando-tests/js_vendor_tc, if related TCs exist.

**Comment by rerobika** [comment]
@yichoi I rebased the PR to the latest master, and visited all the related class tests in the `js_vendor_tc ` repo, but until the inheritance won't be supported, we cannot enable new class related tests.

**Comment by clover2123** [comment]
@rerobika 
Would you wrap the new code related to `class` feature with `ESCARGOT_ENABLE_ES2015` macro when you update it later? We need to encapsulate each es6 feature with the macro now. Thank you.

File: src/parser/ast/ClassBodyNode.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                codeBlock->pushCode(ObjectDefineSetter(ByteCodeLOC(m_loc.index), destIndex, propertyIndex, valueIndex), context, this);
+                context->giveUpRegister(); // for drop property index
+            }
+
+            context->giveUpRegister(); // for drop value index
+        }
+
+        SetObjectInlineCache* inlineCache = new SetObjectInlineCache();
+        codeBlock->m_literalData.pushBack(inlineCache);
+        codeBlock->pushCode(SetObjectPreComputedCase(ByteCodeLOC(m_loc.index), classIndex, PropertyName(ctx->staticStrings().prototype), objIndex, inlineCache), context, this);
```
Comment (clover2123) [inline src/parser/ast/ClassBodyNode.h]: @rerobika Oops.. I found `Context` pointer here similar to [default parameter](https://github.com/pando-project/escargot/pull/198#discussion_r275234353). Would you please fix it in another patch?

**Comment by rerobika** [comment]
@clover2123 The next part of the patch is nearly done, and it's fixed on it.

**Pull request merged**